### ISSUE TITLE
Feature request: print the name of the event

### ISSUE THREAD

**Issue by cyberic99** [untitaker/vdirsyncer#210]
It would be nice, when uploading or removing events or contacts, to display the title of the item, rather than its ID.
For instance, instead of :
Copying (uploading) item 20150521T064448Z-23126@df43914 to calendar_local/xxx
Deleting item 3e54de78-ae48-42b5-8996-998e082724 from contacts_local/xxx

we could have:
Copying (uploading) item "buy stuff" to calendar_local/xxx
Deleting item "Mum" from contacts_local/xxx

**Comment by untitaker** [comment]
I'm open to the idea, here are my thoughts:

* I personally consider the absence of the item's content in vdirsyncer's logs to be a win -- it seems more "discrete" to me, and the user has to worry less about the sensitivity of the logs (unless `-vdebug` is set). Therefore I'd like to rather make this behavior opt-in, if it's going to happen.
* We'd have to figure out a way to restrict the character range vdirsyncer outputs. E.g. items that have `\r` in the SUMMARY (like in https://github.com/untitaker/vdirsyncer/issues/207) would completely break the terminal.
* It probably won't affect performance, though we still should think about it.
* Is there a reasonably popular sync tool which already does that? Which ones *used to* do that, why did they decide to ditch it?

**Comment by cyberic99** [comment]
Hello!
I agree that it should be an opt-in.
And that you would have to filter the display a bit.
But it could be a great addition, if for instance you want to version your events, or keep a track of it.
It could be used with an automatic commit into a git repo, for instance.
> Is there a reasonably popular sync tool which already does that? Which ones used to do that, why did they decide to ditch it?
Yes, there is dropbox, for instance. It is an opt-in as well, but you can have a notification for each change, with the file name and the last action.

**Comment by untitaker** [comment]
>But it could be a great addition, if for instance you want to version your events, or keep a track of it.

Ahh I see, your motivation is to generate more meaningful commit messages?

**Comment by cyberic99** [comment]
> Ahh I see, your motivation is to generate more meaningful commit messages?

Yes. And also to see a reminder of what happened recently to my calendar and to my contacts

**Comment by untitaker** [comment]
>And also to see a reminder of what happened recently to my calendar and to my contacts

I know that wasn't your intention, but this comment scared me off implementing this a bit: I don't want people to use vdirsyncer as a content reader (in any way, even this one), because it might distort the direction and use-cases of this project. I imagine it's instead better (and not hard) to write a script (even shell might suffice for this) that greps for summaries and sorts the results by mtime -- no need to rely on vdirsyncer for that.

**Comment by cyberic99** [comment]
> I don't want people to use vdirsyncer as a content reader

This is your code, you can decide what you want to code or note.
But deciding what people want to do with your software is... strange...

**Comment by untitaker** [comment]
I feared this would come over wrongly, and I'm sorry about that. I don't have a say in how people use my software. Yet at the same time I believe that the way people use software usually shapes the future (and the goals) of said software. Which *usually* leads to feature-creep.

Look at WordPress: It's used as much more than a blogging system, and it barely fulfills these new demands. Plugins (like the ones for ecommerce) basically half-ass a whole CMS on top of WordPress, and nobody really cares, because why set up two pieces of software for your blog and online shop when you can install a massive shitpile of PHP for both?

I know that this is extremely paranoid of me. Your proposal just enhances the human readability of vdirsyncer's output, but at the same time I don't want to encourage the kind of usage I consider wrong.

I promise I will merge a PR that implements your proposal, but I don't have motivation to write this myself.

**Comment by cyberic99** [comment]
No worries, you don't need to apologize, we are only discussing ;-)
You're right about wordpress.
The example usage I gave earlier, with git, was just an example and I'm not even sure I will version my contacts this way.
But... It could be useful to check what happened since the last sync.
It is not a new usage, it just shows a bit more information (which can be sensitive, I agree).

Thx

**Comment by varac** [comment]
Ping, still would like to have this feature.
@untitaker: My take on the above discussion and to software in general is that I'm ok with a default that doesn't show content of changed items, but please let users opt-in displaying more information.
Freedom of choice.

**Comment by untitaker** [comment]
Sure, patches welcome.

On 5 January 2018 12:11:58 GMT+01:00, varac <notifications@github.com> wrote:
>Ping, still would like to have this feature.
>@untitaker: My take on the above discussion and to software in general
>is that I'm ok with a default that doesn't show content of changed
>items, but please let users opt-in displaying more information.
>Freedom of choice.
>
>-- 
>You are receiving this because you were mentioned.
>Reply to this email directly or view it on GitHub:
>https://github.com/pimutils/vdirsyncer/issues/210#issuecomment-355530758

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.

**Comment by Rostiger** [comment]
Hey there - I'd love to see this feature implemented too! 
I'm using multiple calendars shared with different people who also add events to these calendars. The only way to tell if a person added a new event is by looking at the syncing log, which is in turn cumbersome to decipher from the ID.
Any chances this might see the light of day at some point? :)

**Comment by WhyNotHugo** [comment]
vdirsyncer does not parse the `ics` (not beyond reading the UID), so this is rather non-trivial.

**Comment by Rostiger** [comment]
I wrote a bash script that stores vdirsyncer's output in a text file and parses it to obtain information about the calendars and the events and prints them to stdout.
It works well enough for me, maybe it's useful for others too:

https://git.sr.ht/~rostiger/scripts/tree/main/item/clndr
